pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -150,24 +154,23 @@ def get_model_from_plugin(
         Returns: Reference :class:`LightningModule` within parallel wrapper.
 
         """
-        if isinstance(model, LightningDistributedDataParallel):
-            return model.module
+        if isinstance(model, DistributedDataParallel):
+            model = model.module
+        if isinstance(model, LightningDistributedModule):
+            model = model.module
```

Comment:
```suggestion
        if isinstance(model, (DistributedDataParallel, LightningDistributedModule)):
            model = model.module
```

Reply:
@carmocca the way the if statements are layed out there sequentially is intentional, because when we use DDP the lightning module will be wrapped twice, so we have to do `model.module.module` in genreal :)

Reply:
Touché